metaflow/plugins/aws/eks/kubernetes.py

```diff
@@ -44,6 +45,28 @@ class KubernetesKilledException(MetaflowException):
     headline = "Kubernetes Batch job killed"
 
 
+def gen_job_name(flow_name,
+    run_id,
+    step_name,
+    task_id,
+    attempt
+):
+    """ Generate RFC 1123 compatible job name """
+    long_name = "-".join(
+            [
+                flow_name,
```

Comment:
Technically `flow_name` can start with an underscore, which will make the output of this function non-compliant. 

Also, we would need to sanitize the strings for labels as well - we can (at a later stage) make this a utility method.

Reply:
Fixed. Also I realized that the previous version could generate identical names for `HelloFlow` and `Helloflow`, fixed that too